docs/explanation/hardware-requirements.md

```diff
@@ -0,0 +1,160 @@
+
+Source:
+[TCG PC Client Platform TPM Profile Specification for TPM 2.0][tcg_client_PTP_v1p5r14],
+section `4.5.1 NV Storage Requirements`.
+
+### Firmware stack requirements
+
+Based on the experiments with the
+[ms-tpm-20-ref](https://github.com/microsoft/ms-tpm-20-ref) up to date, we
+estimate the reasonable minimal requirements to be:
+- 1 MB of flash,
+- 128 KB of RAM.
+
+During early stages of to projects, we prefer to focus on features, not
+fighting with memory optimization.
+
+## SPI interface
+
+The TwPM is going to be a SPI slave device, and must comply to the SPI clock
+frequency offered by the SPI host on the mainboard. We can look at the required
+SPI frequency from two angles:
+- what TPM specification requires in this area,
+- what SPI host frequencies are offered by the existing mainboards (chipsets).
+
+### Frequency (TPM specification)
+
+TPM must be capable of operating at frequencies 10-24 MHz. However, the maximum
+frequency requirement may be raised in the future, so the target platform
+should have the SPI interface with capability to handle higher frequencies.
+
+Source: [TCG PC Client Specific TPM Interface Specification][tcg_client_v13r27],
+section `6.4.1 Clocking`.
+* [TCG PC Client Platform TPM Profile Specification for TPM 2.0][tcg_client_PTP_v1p5r14],
+
+### Frequency (mainboards/chipsets)
+
+The PCH’s `SPI0` flash controller supports a discrete TPM on the platform
+via its dedicated `SPI0_CS2#` signal. The platform must have no more than 1 TPM.
+
+`SPI0` controller supports accesses to the `SPI0` TPM at approximately
+`17 MHz`, `33 MHz` and `48 MHz` depending on the PCH soft strap. `20 MHz` is the
+reset default, and a valid PCH soft strap setting can override that.
+
+Based on the above, the device must support a SPI clock of at least `20MHz`
+(preferably more - `33 MHz`, and `48 MHz`).
+
+Source:
+
+- [Intel® 300 Series Chipset Family On-Package Platform Controller Hub (PCH) Datasheet, Volume 1 of 2](https://www.intel.com/content/www/us/en/products/docs/chipsets/300-series-chipset-on-package-pch-datasheet-vol-1.html)
+- [Intel® 600 Series Chipset Family Platform Controller Hub Datasheet, Volume 1 of 2](https://edc.intel.com/content/www/us/en/design/ipla/software-development-platforms/client/platforms/alder-lake-desktop/intel-600-series-chipset-family-platform-controller-hub-pch-datasheet-volume/004/spi0-support-for-tpm/)
+
+## LPC interface
+
+### Interrupts
+
+* Service of SIRQ protocol for interrupts
+* Implementation: emulate the set of individual hardware signals using time
+  division multiplexing between frames
+
+Source: [TCG PC Client Platform TPM Profile Specification for TPM 2.0][tcg_client_PTP_v1p5r14],
+section `6.6.1 LPC Interrupts`.
+
+## Timings
+
+### Command Duration
+
+* Command duration MUST be less than 750ms.
+* This duration applies just to a handful of commands, mostly used for
+  initialization, self-testing and RTM:
+
+```text
```

Comment:
This comes from TPM1.2 spec, please remove. TPM2.0 doesn't have `TPM_CAP_PROP_DURATION`.